streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -334,16 +325,22 @@ public void commit() {
      * @throws TaskMigratedException if committing offsets failed (non-EOS)
      *                               or if the task producer got fenced (EOS)
      */
-    private void commitState() {
+    private void commitState(final boolean writeCheckpoint) {
```

Comment:
MInor improvement: we include writing the checkpoint and the caller can indicate if it should be written or not.